updated new functions and tests

Rutgervdspek opened the pull request.

Rutgervdspek committed:
updated new functions and tests

AlexisDerumigny commented on NEWS.md:
```diff
@@ -1,3 +1,14 @@
+* New functions:
+
+  - `CondMeanEst`: nonparametric estimation of the conditional mean.
+  (#2, thanks to Rutger van der Spek)
```
Here, the numbering of the PR is off: you mention 2 but it should be #6 and this one #7

AlexisDerumigny commented on R/EllDistrEst.R:
```diff
@@ -179,14 +187,22 @@ EllDistrEst <- function(X, mu = 0, Sigma_m1 = diag(d),
-              ^ (d/2) ) ^ (2/d) )
-      if (dopb){ pbapply::setpb(pb, i) }
+      for (i in 1:n) {
+        x_quad = Rmpfr::mpfr(as.numeric((X[i,] - mu) %*% Sigma_m1 %*% (X[i,] - mu)),
```
Here I think you can just write 
`x_quad = Rmpfr::mpfr( (X[i,] - mu) %*% Sigma_m1 %*% (X[i,] - mu), `
so without the `as.numeric`

AlexisDerumigny commented on tests/testthat/test-CondCovEst.R:
```diff
@@ -0,0 +1,101 @@
+  gridZ = seq(0, 1, length.out = 4)
+  h = 0.5
+  
+  est = CondCovEst(X, Z, gridZ, h)
```
Could you update this test with the three different types of estimators?

AlexisDerumigny commented on tests/testthat/test-CondEllGenEst.R:
```diff
@@ -0,0 +1,90 @@
+  expect_equal(dim(gEst), c(length(grid), length(gridZ)))
+})
+
+test_that("CondEllGenEst produces finite, non-negative values for simple Gaussian data", {
```
Could you add a test that the mpfr version gives similar results in this simple case?

AlexisDerumigny commented on tests/testthat/test-CondEllGenEst.R:
```diff
@@ -0,0 +1,90 @@
+  
+  mu = rbind(gridZ, gridZ)          
+  sigma = array(0, dim = c(2,2,2))
+  sigma[,,] = diag(2)               
```
Could you also add one more test where the covariance matrix is not the identity? Just to test whether this work also (this is also linked with the standardization of the generator that we discussed, in relation with commit fae2f25cdd1b0e4ccbc1b1210f22ee93d4619d42 )

AlexisDerumigny reviewed (changes requested):
Thanks for the pull request, very nice! I have a few comments that I put in as well.
Could you do the changes, and before this, could you also merge the current main branch in your branch so as to avoid the current merge conflict? Then you can update this PR.

If this is not clear, please feel free to let me know

Rutgervdspek commented:
Hi Alexis,

I updated the test files as described, but didn't change the x_quad line.
The as.numeric is needed because mpfr cannot directly handle a 1x1 matrix
in recent R versions. Converting to numeric ensures a scalar is passed,
preventing warnings when running all tests.

When trying to create a pull request, I'm failing the Github checks.The
first failure isn't clear to me, and the second seems odd since I added
rmpfr, and dopb in the documentation of CondEllGenEst, similar to
EllDistrEst. Do you have any advice on how to fix these issues?

Best,
Rutger

AlexisDerumigny commented:
Hi Rutger,

Thanks for the update. The GitHub checks are done on several machines and OS.
I have just had a look at your new PR #9.

I see that the macOS check fails with the error `Error: The template is not valid. r-lib/actions/v2/setup-r-dependencies/action.yaml (Line: 204, Col: 16): hashFiles('./.github/pkg.lock') failed. Fail to hash files under directory '/Users/runner/work/ElliptCopulas/ElliptCopulas'`
so this is a bug at the level of the GitHub actions themselves, so you don't need to worry about them: I will have a look this myself later to see how to fix this (probably I need to update the checks to the latest version).

Both the Windows and Linux checks fails because of a warning:
````
❯ checking Rd \usage sections ... WARNING
  Undocumented arguments in Rd file 'CondEllGenEst.Rd'
    'mpfr' 'precBits' 'dopb'
````
I have seen that you have added the documentation for these arguments in the R file using the template, which is good. But they are not in the corresponding `.Rd` file. so probably you have forgotten to run `devtools::document()`.
Once you will have done this, you can update #9 (you don't need to start a new pull request each time, you can just continue updating the previous one).
Then this should pass all the checks on Windows and Linux.

Best,
Alexis
